ipa-core/src/protocol/ipa_prf/validation_protocol/proof_generation.rs

```diff
@@ -0,0 +1,517 @@
+        let ctx_left = ctx.set_total_records(length);
+
+        // set up channel
+        let receive_channel_right =
+            &ctx_left.recv_channel::<Fp61BitPrime>(ctx.role().peer(Direction::Right));
+
+        // receive from the right
+        Ok(ctx
+            .parallel_join(
+                (0..length)
+                    .map(|i| async move { receive_channel_right.receive(RecordId::from(i)).await }),
+            )
+            .await?
+            .into_iter()
+            .collect())
+    }
+
+    /// This function generates a batch of quadratic proofs
+    /// it iteratively generates proofs and collects them in a batch
+    ///
+    /// It generates `batch_from_left` as its left output
+    /// and `prover_left_batch` as its right output.
+    /// Further, it outputs the generated masks needed for the verification of proof from the left and right.
+    ///
+    /// Outputs `(batch_from_left, prover_left_batch, (p_mask, q_mask))`
+    ///
+    /// ## Panics
+    /// Panics when the recursion factor is smaller than 2 or masks cannot be set safely (implementation issue).
+    fn compute_proof<C, I>(ctx: &C, uv_tuple_block: I) -> (Self, Self, (Fp61BitPrime, Fp61BitPrime))
+    where
+        C: Context,
+        I: Iterator<Item = UVTupleBlock<Fp61BitPrime>> + Clone,
+    {
+        // set up record counter
+        let mut record_counter = RecordId::from(0);
+
+        // generate first proof from input iterator
+        let (mut uv_values, first_proof_from_left, prover_left_first_proof) = {
+            // precomputation for first proof
+            // first denominator
+            let first_denominator = CanonicalLagrangeDenominator::<
+                Fp61BitPrime,
+                { LargeProofGenerator::RECURSION_FACTOR },
+            >::new();
+            // first lagrange table
+            let first_lagrange_table = LagrangeTable::<
+                Fp61BitPrime,
+                { LargeProofGenerator::RECURSION_FACTOR },
+                { LargeProofGenerator::LAGRANGE_LENGTH },
+            >::from(first_denominator);
+
+            LargeProofGenerator::compute_next_proof(
+                ctx,
+                &mut record_counter,
+                &first_lagrange_table,
+                Self::polynomials_from_blocks(uv_tuple_block),
+            )
+        };
+
+        // storage for other proofs
+        let mut prover_left_proofs =
+            Vec::<[Fp61BitPrime; SmallProofGenerator::PROOF_LENGTH]>::new();
+        let mut proofs_from_left = Vec::<[Fp61BitPrime; SmallProofGenerator::PROOF_LENGTH]>::new();
+
+        // precomputation for other proofs
+        // denominator
+        let denominator = CanonicalLagrangeDenominator::<
+            Fp61BitPrime,
+            { SmallProofGenerator::RECURSION_FACTOR },
+        >::new();
+        // lagrange table
+        let lagrange_table = LagrangeTable::<
+            Fp61BitPrime,
+            { SmallProofGenerator::RECURSION_FACTOR },
```

Comment:
This is basically duplicate code. The const params are also causing the code to wrap and become really unreadable.

Reply:
the duplicate code is resolved. We could shorten the names for the proofgenerator types and associated constants to not cause the code to wrap.